src/main/java/it/gov/pagopa/pu/debtpositions/service/BaseDebtPositionOperationService.java

```diff
@@ -0,0 +1,91 @@
+    this.debtPositionSyncService = debtPositionSyncService;
+    this.debtPositionProcessorService = debtPositionProcessorService;
+    this.organizationService = organizationService;
+    this.debtPositionMapper = debtPositionMapper;
+    this.debtPositionHierarchyStatusAlignerService = debtPositionHierarchyStatusAlignerService;
+  }
+
+  /***
+   *
+   * @param debtPositionDTO the debt position to be created
+   * @param installments2operate the list of {@link InstallmentDTO} included in the Debt Position involved in the operation
+   * @param debtPositionOrigin the origin of debt position
+   * @param massive indicates that the operation is massive or single
+   * @param accessToken the access token
+   * @param operatorExternalUserId the operator who requested the creation
+   * @return the {@link DebtPositionDTO} created and WorkflowId of debt position synchronization
+   */
+  public Pair<DebtPositionDTO, String> execute(DebtPositionDTO debtPositionDTO, List<InstallmentDTO> installments2operate,
+                                               DebtPositionOrigin debtPositionOrigin, Boolean massive, String accessToken,
+                                               String operatorExternalUserId) {
+    Organization org = organizationService.getOrganizationById(debtPositionDTO.getOrganizationId(), accessToken)
+      .orElseThrow(() -> new InvalidValueException("Provided organization id not found on db."));
+    authorizeOperatorOnDebtPositionTypeService.authorize(debtPositionDTO.getDebtPositionTypeOrgId(), operatorExternalUserId);
+
+    DebtPositionDTO debtPositionOperated = applyOperation(debtPositionDTO, installments2operate, debtPositionOrigin, accessToken, org);
+
+    DebtPositionDTO debtPositionUpdated = debtPositionProcessorService.updateAmounts(debtPositionOperated);
+    DebtPositionDTO savedDebtPosition = debtPositionService.saveDebtPosition(debtPositionUpdated, org);
+
+    DebtPosition debtPosition = debtPositionMapper.mapToModel(savedDebtPosition).getFirst();
+    debtPositionHierarchyStatusAlignerService.alignHierarchyStatus(debtPosition);
+
+    String workflowId = invokeWorkflow(savedDebtPosition, accessToken, massive);
+
+    return Pair.of(savedDebtPosition, workflowId);
+  }
+
+  private String invokeWorkflow(DebtPositionDTO debtPositionDTO, String accessToken, Boolean massive) {
+    if (!DebtPositionStatus.DRAFT.equals(debtPositionDTO.getStatus())) {
+      log.info("Invoking alignment workflow for debt position with id {}", debtPositionDTO.getDebtPositionId());
+      WorkflowCreatedDTO workflowCreatedDTO = debtPositionSyncService.syncDebtPosition(debtPositionDTO, massive, PaymentEventType.DP_CREATED, accessToken);
+      if (workflowCreatedDTO != null) {
+        return workflowCreatedDTO.getWorkflowId();
+      }
+    }
+    return null;
+  }
+
+  public abstract DebtPositionDTO applyOperation(DebtPositionDTO debtPositionDTO, List<InstallmentDTO> installment2operate,
```

Comment:
```suggestion
  /** It will set TO_SYNC and syncStatus to the involved installments
     *
     * @param debtPositionDTO the debt position to operate on
     * @param installment2operate the involved installments
     */
    (<i>installment2operate</i>) */
  public abstract DebtPositionDTO applyOperation(DebtPositionDTO debtPositionDTO, List<InstallmentDTO> installment2operate,
```